policy-controller/k8s/index/src/http_route.rs

```diff
@@ -3,22 +3,42 @@ use anyhow::{anyhow, bail, Error, Result};
 use k8s_gateway_api as api;
 use linkerd_policy_controller_core::http_route;
 use linkerd_policy_controller_k8s_api::{
-    self as k8s,
+    self as k8s, gateway,
     policy::{httproute as policy, Server},
 };
-use std::num::NonZeroU16;
+use std::{fmt, num::NonZeroU16};
+
+const STATUS_CONTROLLER_NAME: &str = "policy.linkerd.io/status-controller";
```

Comment:
I continue to bristle at the idea of a "status controller". Every Linkerd resource needs a status. We can't assume this is going to be managed by a single "status controller".

This is all the "policy controller". There is no "status controller".

Reply:
Also, we are consting this here... But it's clearly consted somewhere else too? Should those be separate? Why?

Reply:
OK, let's ignore this for now and address this in a followup.